Themes Showcase: Only display beta badges for FSE beta eligible sites

jeyip opened the pull request:
#### Changes proposed in this Pull Request

We only want to display beta badges when sites are full site editing (FSE) eligible, which is when sites have the `fse-eligible` blog sticker. Otherwise, we'd like to hide the beta badges.

| With Sticker      | Without Sticker |
| --- | --- |
| <img width="1344" alt="Screen Shot 2021-10-08 at 4 54 40 PM" src="https://user-images.githubusercontent.com/5414230/136636474-f53ee026-2e24-4679-b576-c5d5915e599c.png"> | <img width="1344" alt="Screen Shot 2021-10-08 at 5 13 17 PM" src="https://user-images.githubusercontent.com/5414230/136636797-144e4f6f-b3be-4bb5-a3df-142920f200dc.png"> |
| <img width="1344" alt="Screen Shot 2021-10-08 at 4 54 20 PM" src="https://user-images.githubusercontent.com/5414230/136636848-cd794636-8ceb-4bd1-9097-9a04f3eb9671.png"> | <img width="1344" alt="Screen Shot 2021-10-08 at 4 53 59 PM" src="https://user-images.githubusercontent.com/5414230/136636857-20755774-7171-40f3-a239-460ce9122def.png"> | 

#### Testing instructions
<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. [Create a full site editing enabled site](https://horizon.wordpress.com/new?flags=gutenboarding/site-editor)
2. Ensure that it has the `fse-eligible` blog sticker. If not, run `wp blog-stickers add --who=USER --url=SITE_URL --sticker=fse-eligible`. Also, remove the `core-site-editor-enabled` sticker if it exists. I'm unsure if this matters, but we're in the process of removing it entirely from FSE onboarding.
3. Verify that beta badges appear in the theme cards, the current theme card, and the theme info sheet.
4. Run `wp blog-stickers remove --who=USER --url=SITE_URL --sticker=fse-eligible` to make a site FSE beta ineligible
5. Return to the themes showcase
6. Verify that beta badges for theme cards, the current theme card, and the theme info sheet have disappeared.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/56854

jeyip committed:
Add block editor settings higher order component

Although a hook exists to fetch block editor settings, we can't use it in class-based React components. We create a HOC that consumes the useBlockEditorSettingsQuery in order to share the hook's logic with these older components

jeyip committed:
Hide theme card beta badges for FSE ineligible sites

jeyip committed:
Hide theme info sheet beta badge for FSE ineligible sites

jeyip committed:
Hide current theme card beta badge for FSE ineligible sites

jeyip committed:
Hide FSE tab for FSE ineligible sites

zaguiini commented on client/components/theme/index.jsx:
```diff
@@ -275,12 +280,11 @@ export class Theme extends Component {
-		return {
-			isSiteUsingCoreSiteEditor: isSiteUsingCoreSiteEditorSelector( state, siteId ),
-		};
+		return { siteId: getSelectedSiteId( state ) };
```
Do we need to expose `siteId` here? It's not being used and `withBlockEditorSettings` gets it from the store.

jeyip replied:
Ah good catch 👍. Removing the reference now.

jeyip replied:
Updated in ac5b7a6

zaguiini reviewed (approved):
LGTM. 🥇

jeyip committed:
Remove unnecessary siteId reference

vindl commented on client/data/block-editor/with-block-editor-settings.js:
```diff
@@ -0,0 +1,16 @@
+const withBlockEditorSettings = createHigherOrderComponent(
+	( Wrapped ) => ( props ) => {
+		const siteId = useSelector( getSelectedSiteId );
+		const { data } = useBlockEditorSettingsQuery( siteId );
```
Should we pass this directly without creating the HOC?

jeyip replied:
That was my original approach, but we needed to inject this data into legacy class-based components, and this seemed like the most straightforward way to do it.

vindl commented:
The unit test failures look related to this PR. We should resolve them before merging.

<img width="712" alt="Screenshot 2021-10-13 at 13 00 30" src="https://user-images.githubusercontent.com/1182160/137120871-3c10c2f6-afe5-42d0-8376-b9d7486e5632.png">

jeyip committed:
Fix logged-out test

jeyip committed:
Disable settings query when site id does not exist

When clients are logged out, there is no site id to associate with the session. Because of this, React query in the useBlockEditorSettings hook will make continuous requests unless disabled.

jeyip committed:
Refactor logged-out test to account for react query

jeyip committed:
Update themes-list dom snapshot

jeyip committed:
Refactor theme main test to account for react query

zaguiini commented:
> @vindl All specs should be updated.
> 
> @zaguiini I refactored `client/my-sites/theme/test/main.jsx` in a similar way as to what we did with `logged-out`, but I didn't switch in-memory rendering over to `react-testing-library` yet. It was throwing method not available errors for the jsdom document object and type errors for wpcom specific methods attached to the window, the latter of which seems like it may be a rabbit hole 😬
> 
> I was thinking that, if we really want to continue refactoring, we could handle those in a follow-up. Thoughts?

Sure, let's do it on another opportunity. Not a priority, I think. It still works 🤷‍♂️

jeyip commented:
E2E test failures appear unrelated to this PR. Rerunning them to make sure.

jeyip committed:
Fix proptype declarations

jeyip committed:
Add react query provider and query client to server side rendered layout

jeyip committed:
Only enable react query if user is logged in

jeyip commented on client/controller/index.node.js:
```diff
@@ -31,9 +33,15 @@ const ProviderWrappedLoggedOutLayout = ( {
-			<ReduxProvider store={ store }>
-				<LayoutLoggedOut primary={ primary } secondary={ secondary } redirectUri={ redirectUri } />
-			</ReduxProvider>
+			<QueryClientProvider client={ queryClient }>
```
@jsnajdr I added a `QueryClientProvider` and `QueryClient` here as suggested p1634208328117100/1634170889.108700-slack-CHN6J22MP. I'm planning to add an inline comment in a little bit.

jeyip commented on client/data/block-editor/use-block-editor-settings-query.ts:
```diff
@@ -6,14 +6,19 @@ export type BlockEditorSettings = {
 export const useBlockEditorSettingsQuery = (
-	siteId: string
+	siteId: string,
+	userLoggedIn = false
```
I added a param to check whether or not a user is logged in for the block editor settings hook. Just wanted to let you know for visibility.

@zaguiini

jeyip committed:
Add inline comment to server side rendered ProviderWrappedLoggedOutLayout
